Adding support for setting tmux window titles

smeetrs opened the pull request:
This PR adds support for updating tmux window titles as described in #4.

Steps to verify the functionaity:

1. Install tmux.
2. Create tmux configuration file and allow applications to set tmux window titles.
```
echo "set -g allow-rename on" >> ~/.tmux.conf
```
3. Start tmux.
```
tmux new-session
```
4. Start zsh if it isn't the default shell.
5. Tmux window title should be dynamically updated based on current path and running command (refer screenshot).
![image](https://user-images.githubusercontent.com/29375430/173185995-cc5fedc4-af27-4c6f-8b3e-afcc8bbde337.png)
6. Exit shell to close tmux.

olets commented:
Hi @lordmartian I've been very busy. Don't take my silence as dismissal. I'll be afk for a couple weeks, but in a month or maybe sooner I should be free to check this out. Were you still interested in getting it merged? Or maybe you closed the PR because it doesn't work?

smeetrs commented:
The closing of PR was unintentional. I had deleted my fork (as I moved the title handling code to my zshrc instead of using a plugin) and that may have caused it to get closed. The changes in the PR work as expected. I don't see any option to re-open it from my end. Is it possible from yours?

In any case, there are only a few changes and can be easily merged manually if needed.

p1r473 commented:
I took a stab at maximum compatability here: https://github.com/olets/zsh-window-title/pull/7
Make sure your ~/.tmux.conf contains
```
set -g allow-rename on
set -g set-titles on
set -g set-titles-string '#T'
set -g automatic-rename off
```

olets commented:
@smeetrs time flies! Thanks for the contribution.

@p1r473 thanks again for sharing your take. Helped me to understand tmux and figure out which direction I wanted to go.
